.cirrus.yml

```diff
@@ -0,0 +1,251 @@
+env:
+  LANG: C
+  DEFAULT_CONFIG_OPTS: --enable-debug --without-libstatgrab
+
+
+###
+# make distcheck and other sanity checks
+#
+relase_ready_task:
+  container:
+    image: collectd/ci:stretch_amd64
+  lint_script:
+    - /checks/check-bashisms.sh
+    - /checks/check-pod.sh
+  configure_script:
+    - ./build.sh
+    - ./configure $DEFAULT_CONFIG_OPTS
+  checks_script:
+    - make -s distcheck
+
+###
+# Default toolchain and build flags use in deb packages, on a range of Debian
+# and Ubuntu releases (+ Debian/unstable)
+# Most should succeed, and PRs shouldn't break them.
+#
+debian_default_toolchain_task:
+  matrix:
+    - allow_failures: true
+      skip_notifications: true
+      container:
+        image: collectd/ci:wheezy_amd64 # TODO: fix this platform
+    - allow_failures: false
+      container:
+        image: collectd/ci:jessie_amd64
+    - allow_failures: false
+      container:
+        image: collectd/ci:stretch_amd64
+    - allow_failures: false
+      container:
+        image: collectd/ci:stretch_i386
+    - allow_failures: false
+      container:
+        image: collectd/ci:trusty_amd64
+    - allow_failures: false
+      container:
+        image: collectd/ci:xenial_amd64
+    # debian/unstable is expected to fail
+    - allow_failures: true
+      skip_notifications: true
+      only_if: $CIRRUS_BRANCH == 'master'
+      container:
+        image: collectd/ci:sid_amd64
+  configure_script:
+    - ./build.sh
+    - gcc --version
+    - >
+      ./configure CC=gcc $DEFAULT_CONFIG_OPTS
+      CFLAGS="$(dpkg-buildflags --get CFLAGS)"
+      CPPLAGS="$(dpkg-buildflags --get CPPFLAGS)"
+      LDFLAGS="$(dpkg-buildflags --get LDFLAGS)"
+  build_script:
+    - make -sk
+  tests_script:
+    - make -sk check
```

Comment:
do make distcheck here too? that will catch a few more things sooner.

Reply:
My understanding is that `make distcheck` is equivalent to running `make install && make uninstall` into a temporary directory, right ? In this case, the platform doesn't really make a different, or does it ?

Unlike the jenkins-based CI, here we run `release_ready_task` on any branch, PRs or not. So a `make distcheck` is always run.

So unless my understanding is incorrect, running `distcheck` more than once wouldn't bring anything. But happy to change this if I'm wrong ;-)

Reply:
Yes it is equivalent. It could matter if make installs behaves differently on a different platform, for instance if we install specific files for FreeBSD, but I don't think we do.